python/pyarrow/_compute.pyx

```diff
@@ -1294,3 +1294,30 @@ class TDigestOptions(_TDigestOptions):
         if not isinstance(q, (list, tuple, np.ndarray)):
             q = [q]
         self._set_options(q, delta, buffer_size, skip_nulls, min_count)
+
+
+def _group_by(args, keys, aggregations):
```

Comment:
We can also make this a public function in the compute module?

Reply:
Not sure, should we? I made it internal because we plan to replace this with the exec engine on long term, so I guess that the `Table.group_by` implementation will switch to use something different in the future.

Reply:
> I made it internal because we plan to replace this with the exec engine on long term, so I guess that the `Table.group_by` implementation will switch to use something different in the future.

The same could be done for a `pyarrow.compute` function? (it doesn't map 1:1 to a C++ kernel anyway)

For me one reason to put it in the compute functions as a `pc.group_by(table, keys, ...)` is to sidestep the 1-step vs 2-step API discussion for the method a bit. For a function in compute, I think it's totally fine to be a one step function